### ISSUE TITLE
Invocation trashes symlink target instead of symlink itself

### ISSUE THREAD

**Issue by uranusjr** [ArturKovacs/trash-rs#22]
Something like

```console
$ touch x    # Create file 'x'
$ ln -s x y  # Create a symbolic link of 'x' named 'y'
```

```rust
// Rust code.
trash::remove("y");
```

```console
$ ls
y
```

The original file `x` is removed instead of the symbolic link `y`.

I took a look at the implementation, this is likely caused by the `canonicalize()` call on the input. This function resolves symbolic links according to Rust documentation. I think the solution would be to remove that call, and if absolute paths should be constructed by joining a relative path to `current_dir()`—although my gut feeling is they may not be needed at all; OS APIs tend to work with relative paths as-is.

**Comment by ArturKovacs** [comment]
The `SHFileOperationW` used by the Windows implementation requires "fully qualified paths"; in other words the path needs to be absolute there and I believe that's why I used the `canonicalize()` call which ended up in all implementations for consistency I would say.

So the solution for the Windows platform could be to only `canonicalize()` the parent folder path and then join the `file_name()` part to the canonicalized parent path. This method could be used on every platform for consistency however I believe, simply passing the relative path has the same effect to this, on macOS and Linux, although I'm not sure.

**Comment by ArturKovacs** [comment]
I added a new pair of functions namely `delete` and `delete_all` that delete the symlink instead of the link destination. It's available at ba850eee27299e2aca0b1fc634f566f91a40e43b. Could you please check and confirm this is what you needed, @uranusjr?

**Comment by uranusjr** [comment]
I just tested HEAD on macOS and it works as expected.

**Comment by ArturKovacs** [comment]
The new functions are now released with `v1.2`
